vector/bool.go

```diff
@@ -28,7 +28,7 @@ func (b *Bool) Type() zed.Type {
 }
 
 func (b *Bool) Value(slot uint32) bool {
-	return (b.Bits[slot>>6] & (1 << (slot & 0x3f))) != 0
+	return b != nil && (b.Bits[slot>>6]&(1<<(slot&0x3f))) != 0
```

Comment:
I did this because we were getting panics in vector.StringValue when the Null boolean vector was nil. I could change the code there to check for nulls but it seemed to make more sense to just put the check here.

Reply:
I think all the vector types should behave the same way with regard to nil receiver and my inclination here is to check at the caller and assume b not nil. Since we're here though, I noticed that Value should check for the nulls vector as in other vector types.

Reply:
Hmm after looking deeper I think you're change is ok but include a comment about the nil receiver for Bool.Value() specifically referring to the Nulls use case.